Client/Frontend/Browser/New Tab Page/Sections/FavoritesOverflowSectionProvider.swift

```diff
@@ -0,0 +1,103 @@
+import BraveUI
+import Data
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+class FavoritesOverflowButton: SpringButton, Themeable {
+    private let backgroundView = UIVisualEffectView(effect: UIBlurEffect(style: .light)).then {
+        $0.clipsToBounds = true
+        $0.isUserInteractionEnabled = false
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        let label = UILabel().then {
+            $0.text = Strings.NTP.showMoreFavorites
+            $0.appearanceTextColor = .white
+            $0.font = UIFont.systemFont(ofSize: 12.0, weight: .medium)
+        }
+        
+        addSubview(backgroundView)
+        backgroundView.contentView.addSubview(label)
+        
+        backgroundView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        label.snp.makeConstraints {
+            $0.edges.equalToSuperview().inset(UIEdgeInsets(top: 5, left: 10, bottom: 5, right: 10))
+        }
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        backgroundView.layer.cornerRadius = bounds.height / 2.0 // Pill shape
+    }
+}
+
+class FavoritesOverflowCell: NewTabCollectionViewCell<FavoritesOverflowButton> {
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        view.snp.remakeConstraints {
+            $0.top.bottom.equalToSuperview()
+            $0.centerX.equalToSuperview()
+        }
+    }
+    
+    override func preferredLayoutAttributesFitting(_ layoutAttributes: UICollectionViewLayoutAttributes) -> UICollectionViewLayoutAttributes {
+        // swiftlint:disable:next force_cast
+        let attributes = layoutAttributes.copy() as! UICollectionViewLayoutAttributes
+        attributes.size.height = systemLayoutSizeFitting(layoutAttributes.size, withHorizontalFittingPriority: .required, verticalFittingPriority: .fittingSizeLevel).height
+        return attributes
+    }
+}
+
+class FavoritesOverflowSectionProvider: NSObject, NTPSectionProvider {
+    let action: () -> Void
+    
+    private var frc: NSFetchedResultsController<Bookmark>
+    private var count: Int = 0
+    
+    init(action: @escaping () -> Void) {
+        self.action = action
+        frc = Bookmark.frc(forFavorites: true, parentFolder: nil)
+        super.init()
+        
+        do {
+            count = try frc.managedObjectContext.count(for: frc.fetchRequest)
+        } catch {
+            log.error("Favorites fetch error")
+        }
+    }
+    
+    @objc private func tappedButton() {
+        action()
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let width = fittingSizeForCollectionView(collectionView, section: section).width
+        return count > FavoritesSectionProvider.numberOfItems(in: collectionView, availableWidth: width) ? 1 : 0
+    }
+    func registerCells(to collectionView: UICollectionView) {
```

Comment:
Can we put new lines after a function and before the new one? Just seems a little cluttered is all.

Reply:
Sure

Reply:
Done